bynary/apps/demo/src/app/core/feature/recipe/pages/detail/recipe-detail.component.spec.ts

```diff
@@ -61,87 +76,107 @@ describe('RecipeDetailComponent', () => {
-            expect(recipesFacadeStub.deleteRecipe).toHaveBeenCalledTimes(1);
+        describe('onDeleteRecipe', () => {
+
+            beforeEach(() => {
+                component.onDeleteRecipe();
+            });
+
+            it('should call recipeFacade.deleteRecipe with the correct arguments', () => {
+                expect(mockedRecipeFace.deleteRecipe).toHaveBeenCalledWith(1);
+            });
+
+            it('should call router.navigate with the correct arguments', () => {
+                expect(mockedRouter.navigate).toHaveBeenCalledWith(['recipes']);
+            });
 
-            router.navigate(['recipes']);
-            expect(router.navigate).toHaveBeenCalledWith(['recipes']);
         });
 
     });
 
-
     describe('template', () => {
         let debugElement: DebugElement;
 
         beforeEach(() => {
             debugElement = fixture.debugElement;
         });
 
-        it('should display recipe with name, image, description and ingredients', () => {
-            const recipeImage = debugElement.query(By.css('.recipe-image')).properties;
-            const recipeName = debugElement.query(By.css('.recipe-name'));
-            const recipeDescribtion = debugElement.query(By.css('.recipe-description'));
-            const recipeIngredients = debugElement.queryAll(By.css('.recipe-ingredients'));
+        describe('recipe', () => {
 
-            expect(recipeImage.src).toEqual('https://upload.wikimedia.org/wikipedia/commons/b/b9/Recipe_icon.svg');
-            expect(recipeName.nativeElement.textContent).toEqual('Schnitzel mit Pommes');
-            expect(recipeDescribtion.nativeElement.textContent).toEqual(' Bla bla ');
+            it('should have a name', () => {
+                const recipeName = debugElement.query(By.css('.c-recipe-detail-component__recipe-name'));
+
+                expect(recipeName.nativeElement.textContent).toEqual('Schnitzel mit Pommes');
+            });
+
+            it('should have an image', () => {
+                const recipeImage = debugElement.query(By.css('.c-recipe-detail-component__recipe-image')).properties;
+
+                expect(recipeImage.src).toEqual('https://upload.wikimedia.org/wikipedia/commons/b/b9/Recipe_icon.svg');
+            });
+
+            it('should have a description', () => {
+                const recipeDescription = debugElement.query(By.css('.c-recipe-detail-component__recipe-description'));
+
+                expect(recipeDescription.nativeElement.textContent).toEqual(' Bla bla ');
+            });
+
+            it('should have a ingredients', () => {
+                const recipeIngredients = debugElement.queryAll(By.css('.c-recipe-detail-component__recipe-ingredients'));
+
+                expect(recipeIngredients).toHaveLength(2);
+                expect(recipeIngredients[0].nativeElement.textContent).toEqual(' Schnitzel - 2 ');
+                expect(recipeIngredients[1].nativeElement.textContent).toEqual(' Pommes - 1 ');
```

Comment:
Hier könntest du es vllt noch einmal in einen Describe-Block schreiben und dort dann getrennt beide Ingredients abfragen.
```suggestion
            describe('ingredients', () => {
              let recipeIngredients: DebugElement[];
              
              beforeEach(() => {
                  recipeIngredients  = debugElement.queryAll(...)
              })
              
              it('should have 2 ingredients'),
              it('should have the correct first ingredient')
              ...
            
            })
 
```